python_modules/dagster/dagster/core/definitions/partition.py

```diff
@@ -168,17 +170,19 @@ def get_default_partition_mapping(self):
         return IdentityPartitionMapping()
 
 
+@whitelist_for_serdes
 class StaticPartitionsDefinition(
-    PartitionsDefinition[str]
-):  # pylint: disable=unsubscriptable-object
-    def __init__(self, partition_keys: List[str]):
+    PartitionsDefinition[str],  # pylint: disable=unsubscriptable-object
+    NamedTuple("_StaticPartitionsDefinition", [("partition_keys", List[str])]),
+):
+    def __new__(cls, partition_keys: List[str]):
         check.list_param(partition_keys, "partition_keys", of_type=str)
-        self._partitions = [Partition(key) for key in partition_keys]
+        return super(StaticPartitionsDefinition, cls).__new__(cls, partition_keys)
 
     def get_partitions(
         self, current_time: Optional[datetime] = None  # pylint: disable=unused-argument
     ) -> List[Partition[str]]:
-        return self._partitions
+        return [Partition(key) for key in self.partition_keys]
```

Comment:
why the switch here?  We can avoid constructing all these Partition objects over and over again in the static case.

Reply:
In order to make this class a `NamedTuple` in order to serialize it, the partition keys must be an instance variable instead of the partitions objects because the partition keys are accepted as an argument upon instantiation. If we want to avoid this, I can do something similar to the time based partition case and create an `ExternalStaticPartitionsData` object that serializes the partitions list